### ISSUE TITLE
Chromium Just Waits Forever On First Sync

### ISSUE THREAD

**Issue by evmcl** [xBrowserSync/App#40]
Hi,

Chromium Version 69.0.3497.100 (Official Build) Arch Linux (64-bit)
xBrowserSync extension 1.4.0 from the store

When trying to create a new sync, it just waits forever with the "Do not close this window" message.

The background.html page for the extension has the following errors:

```
background.html:1 Error in response to bookmarks.getSubTree: TypeError: Cannot read property '0' of undefined
    at Object.callback (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3327:30)
    at getLocalBookmarkTreeById (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3326:30)
    at Object.getBookmarks [as Get] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2878:29)
    at sync_handlePush (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2320:47)
    at sync (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2136:27)
    at Object.queueSync [as Sync] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:1879:9)
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3673:30
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:140:44
background.html:1 Unchecked runtime.lastError while running bookmarks.getSubTree: Can't find bookmark for id.
    at getLocalBookmarkTreeById (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3326:30)
    at Object.getBookmarks [as Get] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2878:29)
    at sync_handlePush (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2320:47)
    at sync (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2136:27)
    at Object.queueSync [as Sync] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:1879:9)
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3673:30
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:140:44
```

**Comment by evmcl** [comment]
No worries. Let me know if there's anything else I can provide to help debug.

P.S. and congrats. Hope everything goes smooth.

**Comment by nero120** [comment]
Hi @evmcl, after some investigation I can't see why `getLocalBookmarkTreeById` would be being called with an id that does not exist (only the root children folders are called here, i.e. Bookmarks bar:2 and Other bookmarks:1). As I do not have Arch Linux, I've prepared a [beta release](https://github.com/xBrowserSync/App/releases/tag/v1.4.1_BETA) that contains additional logging in `getLocalBookmarkTreeById` to reveal the local bookmark id that is being requested when the error occurs.

Could I possibly ask you to install the .crx file attached to the release, enable debug mode in xBrowserSync settings, then try creating a release? You can then copy the debug mode message log out of xBrowserSync or grab the output from the background.html console.

Please paste here so I can understand what's going on! 😄

**Comment by evmcl** [comment]
Hi.

I've run as requested. The output is below.

On a lark, I tried running with a new profile and it worked fine, so there's something in my existing profile that is causing the issue. I will do some further analysis on my end.

In the meantime, here is the output from when it fails:

```
background.min.js:1395 getLocalBookmarkTreeById: 2
background.min.js:1395 getLocalBookmarkTreeById: 1
background.html:1 Error in response to bookmarks.getSubTree: TypeError: Cannot read property '0' of undefined
    at Object.callback (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3326:30)
    at getLocalBookmarkTreeById (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3325:30)
    at Object.getBookmarks [as Get] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2876:29)
    at sync_handlePush (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2318:47)
    at sync (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2134:27)
    at Object.queueSync [as Sync] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:1877:9)
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3672:30
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:140:44
background.html:1 Unchecked runtime.lastError while running bookmarks.getSubTree: Can't find bookmark for id.
    at getLocalBookmarkTreeById (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3325:30)
    at Object.getBookmarks [as Get] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2876:29)
    at sync_handlePush (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2318:47)
    at sync (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:2134:27)
    at Object.queueSync [as Sync] (chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:1877:9)
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:3672:30
    at chrome-extension://lcbjdhceifofjlpecfpeimnnphbcjgnc/js/background.min.js:140:44
```

**Comment by nero120** [comment]
@evmcl ok so that's failing when using the chromium API to get the bookmarks sub tree for "Bookmarks bar" (id 1). Do you have any bookmarks in the bookmarks bar, and does it appear under chrome://bookmarks like so:

![image](https://user-images.githubusercontent.com/4236238/47827430-cd4ba900-dd75-11e8-8cd6-633382693a5d.png)

**Comment by evmcl** [comment]
Yes, I have a fair number of bookmarks on my bookmarks bar. The folder structure looks like this:

![bookmarks](https://user-images.githubusercontent.com/46415/47827897-7a550e80-ddd4-11e8-9db4-8bfa9e200401.png)

**Comment by nero120** [comment]
Yeah that looks fine, can't for the life of me understand why the native bookmarks API is saying `Can't find bookmark for id` when it's being passed the native ids for Bookmarks bar and Other bookmarks!

Are you able to create a backup in xBrowserSync Settings (you don't have to be synced to create a backup)? If so, would you be willing to [email it to me](mailto:admin@xbrowsersync.org) so I can debug what's happening - could be something in your bookmarks as you mentioned it works with a new profile.

**Comment by evmcl** [comment]
I think I've come up with a minimum viable repeat of the issue. Attached is the Bookmarks file from my profile (zipped). I deleted all my bookmarks so this only has the three basic root folders.

If you create a new profile, then copy this Bookmarks file into it, then try installing xBrowserSync and syncing you'll reproduce the error.

[Bookmarks.zip](https://github.com/xBrowserSync/App/files/2545457/Bookmarks.zip)

**Comment by nero120** [comment]
@evmcl that backup shows the other bookmarks id as "4656" when it should be "2". As far as I'm aware, both the bookmarks bar and other bookmarks root nodes cannot be modified so I'm not sure how you ended up in that situation but it looks corrupted to me. Are you able to get a fresh profile to get into this state?

Also, how did you generate that JSON backup file from Chromium? I thought it could only export as HTML?

**Comment by evmcl** [comment]
> how did you generate that JSON backup file from Chromium?

I didn't. I just copied the `~/.config/chromium/Default/Bookmarks` file directly out of my profile.

> Are you able to get a fresh profile to get into this state?

As I was using my bookmarks in Firefox as my canonical copy, I just deleted the above Bookmarks file and synchronised, so I am all up and running now. Feel free to close this issue if you think there is nothing further to be done.

Thanks for all your help.
